src/streamlink/plugins/trovo.py

```diff
@@ -0,0 +1,167 @@
+"""
+$description Global video game live streaming and video hosting platform, owned by Tencent.
+$url trovo.live
+$type live, vod
+"""
+
+import logging
+import random
+import re
+
+from streamlink.plugin import Plugin, pluginmatcher
+from streamlink.plugin.api import validate
+from streamlink.stream.hls import HLSStream
+
+log = logging.getLogger(__name__)
+
+
+@pluginmatcher(re.compile(
+    r"https?://(?:www\.)?trovo\.live/([^/?&]+)(?:/([^/?&]+))?"
```

Comment:
The capture groups could be named and the distinction between live streams and vods/clips be improved, but it should be fine as it is now.

Reply:
I think I've made it better...